packages/@glimmer-workspace/integration-node-tests/test/prettier.test.ts

```diff
@@ -0,0 +1,41 @@
+import { createRequire } from 'node:module';
+
+import * as prettier from 'prettier';
+import { describe, expect, it } from 'vitest';
+
+const require = createRequire(import.meta.url);
+
+/**
+ * See: https://github.com/glimmerjs/glimmer-vm/issues/1688
+ *
+ * Requires the root package.json#pnpm#overrides point at our internal
+ * copy of @glimmer/syntax, or else prettier brings its own already published
+ * copy of @glimmer/syntax
+ *
+ * NOTE: that this test alone is insufficient to test our built outputs.
+ *       the smoke-tests/* folders are for that purpose.
+ */
+describe('Prettier', () => {
+  it(`SMOKE: we've symlinked to the in-repo copy of @glimmer/syntax`, () => {
+    let workspacePath = require.resolve('@glimmer/syntax');
+    let prettierPath = require.resolve('prettier');
+    let prettierGlimmer = require.resolve('@glimmer/syntax', { paths: [prettierPath] });
+
+    expect(prettierGlimmer).toBe(workspacePath);
+  });
+
+  it('Underlynig preprocess API works', async () => {
+    let result = (await import('@glimmer/syntax')).preprocess('<h1></h1>');
+
+    expect(result, `It can be await import()'d, and doesn't error`).toBeTruthy();
+  });
+
+  it('Prettier can call preprocess', async () => {
+    let result = await prettier.format(`     <div>\n</div>`, { parser: 'glimmer' });
```

Comment:
This PR has been settled for a long time, sorry that I didn't see this earlier, but this test is meaningless, since Prettier uses a bundled version of `@glimmer/syntax`.

If we want a reliable test, we need

1. Clone prettier/prettier
2. Upgrade `@glimmer/*` to use the changed version
3. Run Prettier test suite.